### DISCUSSION TITLE
$env.LAST_EXIT_CODE behaviour changed with 0.98.0

### DISCUSSION THREAD

**Discussion by JanDeDobbeleer** [nushell/nushell#13949] [Q&A]
I was looking at the latest release and I can't make `$env.LAST_EXIT_CODE` work the way it used to, regardless of how I configure `display_errors`. I have no idea how I can capture the last exit code of a command passed by the user to correctly use inside oh-my-posh. With the [following prompt configuration](https://github.com/JanDeDobbeleer/oh-my-posh/blob/beta/src/shell/scripts/omp.nu), `$env.LAST_EXIT_CODE` is always zero. I did try to get the value on closure start, but that doesn't make a difference.

**Comment by ghost** [comment]
same here (also with `nu --no-config`). this is most likely a bug and should therefore get a issue

EDIT:
* external commands (`^false`, `git revert`, etc) and `exit` seem to be unaffected
* something in 3d008e2c4e1248672763ec5141146eaaba526b0d broke it

**Comment by fdncred** [comment]
@IanManske do you have any input that can help here?

We've landed a few PRs since the release that could help. My initial advice is to try the latest main to see if it makes any difference. It could be that we haven't found all the cases to fix yet though.

**Reply by ghost** [reply]
its still present in git-head ( 8200831b07ab87ab23c4a56a8a08251d0332b114 ) and was introduced in 3d008e2

**Comment by IanManske** [comment]
Hmm, I'm not able to reproduce this. I tried on 0.98.0 and latest main.

Note: I do not have `oh-my-posh` installed, and I do not have the `SET_POSHCONTEXT` env variable set.

I've used the config file provided by @JanDeDobbeleer, and changed line 12 to `let _omp_executable: string = 'echo'` [^1]. Then I ran `nu --config config.nu --env-config env.nu` (env.nu is a blank file), and this is what I get:

```nu
primary --save-cache --shell=nu --shell-version=0.98.0 --status=0 --no-status=true --execution-time=-1 --terminal-width=248 --cleared=false
# added new lines here for readability
git revert
fatal: not a git repository (or any of the parent directories): .git
primary --save-cache --shell=nu --shell-version=0.98.0 --status=128 --no-status=false --execution-time=3 --terminal-width=248 --cleared=false
```

Same thing with `^false`:
```nu
print primary --save-cache --shell=nu --shell-version=0.98.0 --status=0 --no-status=false --execution-time=1 --terminal-width=248 --cleared=false
^false
print primary --save-cache --shell=nu --shell-version=0.98.0 --status=1 --no-status=false --execution-time=3 --terminal-width=248 --cleared=false
```

[^1]:  I also added `$env.config.display_errors.exit_code = false`, but I tested without this line as well. This should not affect behavior, it only affects whether an error is printed (errors happen the same way regardless of this setting).

**Reply by JanDeDobbeleer** [reply]
@IanManske do you also get an exit code when using a command that doesn't exist? This was my "simple" test, should've added that.

**Reply by IanManske** [reply]
Oh ok, yeah I see the issue now. It might be a little late for a patch release, but this can be fixed for the next release (Oct. 15).

I have a fix up in #13954, but you might also like #13953.

**Reply by JanDeDobbeleer** [reply]
@IanManske perfect, thanks for the blazing fast execution!
